src/mscorlib/src/System/Number.CoreCLR.cs

```diff
@@ -2,307 +2,44 @@
-    // to all non-zero values. If the number to be formatted is non-zero, but
-    // becomes zero after rounding according to the format in the first or second
-    // section, then the resulting zero is formatted according to the third
-    // section.
-    //
-    // For both standard and user-defined formatting operations on values of type
-    // float and double, if the value being formatted is a NaN (Not
-    // a Number) or a positive or negative infinity, then regardless of the format
-    // string, the resulting string is given by the NaNSymbol,
-    // PositiveInfinitySymbol, or NegativeInfinitySymbol property of
-    // the NumberFormatInfo used to format the number.
-    //
-    // Parsing
-    //
-    // The Parse methods provided by the numeric classes are all of the form
-    //
-    //  public static XXX Parse(String s);
-    //  public static XXX Parse(String s, int style);
-    //  public static XXX Parse(String s, int style, NumberFormatInfo info);
-    //
-    // where XXX is the name of the particular numeric class. The methods convert a
-    // string to a numeric value. The optional style parameter specifies the
-    // permitted style of the numeric string. It must be a combination of bit flags
-    // from the NumberStyles enumeration. The optional info parameter
-    // specifies the NumberFormatInfo instance to use when parsing the
-    // string. If the info parameter is null or omitted, the numeric
-    // formatting information is obtained from the current culture.
-    //
-    // Numeric strings produced by the Format methods using the Currency,
-    // Decimal, Engineering, Fixed point, General, or Number standard formats
-    // (the C, D, E, F, G, and N format specifiers) are guaranteed to be parseable
-    // by the Parse methods if the NumberStyles.Any style is
-    // specified. Note, however, that the Parse methods do not accept
-    // NaNs or Infinities.
     internal static partial class Number
     {
+        public static unsafe void DoubleToNumber(double value, int precision, ref NumberBuffer number)
+        {
+            fixed (NumberBuffer* numberPtr = &number)
+            {
+                DoubleToNumber(value, precision, (byte*)numberPtr);
+            }
+        }
         [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern string FormatDouble(double value, string format, NumberFormatInfo info);
+        private static extern unsafe void DoubleToNumber(double value, int precision, byte* number);
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern string FormatSingle(float value, string format, NumberFormatInfo info);
 
-        public static unsafe bool NumberBufferToDecimal(ref Number.NumberBuffer number, ref decimal value)
+        public static unsafe double NumberToDouble(ref NumberBuffer number)
         {
-            fixed (Number.NumberBuffer* numberPtr = &number)
+            fixed (NumberBuffer* numberPtr = &number)
```

Comment:
This wrapper is unnecessary - the signature of the FCall can be: `public static double NumberToDouble(ref NumberBuffer number)`.

Reply:
Fixed.  I removed all of the wrappers.